testing/can-messages/src/messages.rs

```diff
@@ -978,6 +982,375 @@ impl Into<f32> for DolorOneFloat {
     }
 }
 
+/// MultiplexTest
+///
+/// - ID: 200 (0xc8)
+/// - Size: 8 bytes
+/// - Transmitter: SENSOR
+#[derive(Clone, Copy)]
+pub struct MultiplexTest {
+    raw: [u8; 8],
+}
+
+impl MultiplexTest {
+    pub const MESSAGE_ID: u32 = 200;
+
+    pub const MULTIPLEXOR_MIN: u8 = 0_u8;
+    pub const MULTIPLEXOR_MAX: u8 = 2_u8;
+    pub const UNMULTIPLEXED_SIGNAL_MIN: u8 = 0_u8;
+    pub const UNMULTIPLEXED_SIGNAL_MAX: u8 = 4_u8;
+    pub const MULTIPLEXED_SIGNAL_ZERO_A_MIN: f32 = 0_f32;
+    pub const MULTIPLEXED_SIGNAL_ZERO_A_MAX: f32 = 3_f32;
+    pub const MULTIPLEXED_SIGNAL_ZERO_B_MIN: f32 = 0_f32;
+    pub const MULTIPLEXED_SIGNAL_ZERO_B_MAX: f32 = 3_f32;
+    pub const MULTIPLEXED_SIGNAL_ONE_A_MIN: f32 = 0_f32;
+    pub const MULTIPLEXED_SIGNAL_ONE_A_MAX: f32 = 6_f32;
+    pub const MULTIPLEXED_SIGNAL_ONE_B_MIN: f32 = 0_f32;
+    pub const MULTIPLEXED_SIGNAL_ONE_B_MAX: f32 = 6_f32;
+
+    /// Construct new MultiplexTest from values
+    pub fn new(multiplexor: u8, unmultiplexed_signal: u8) -> Result<Self, CanError> {
+        let mut res = Self { raw: [0u8; 8] };
+        res.set_multiplexor(multiplexor)?;
+        res.set_unmultiplexed_signal(unmultiplexed_signal)?;
+        Ok(res)
+    }
+
+    /// Access message payload raw value
+    pub fn raw(&self) -> &[u8] {
+        &self.raw
+    }
+
+    /// Get raw value of Multiplexor
+    ///
+    /// - Start bit: 0
+    /// - Signal size: 4 bits
+    /// - Factor: 1
+    /// - Offset: 0
+    /// - Byte order: LittleEndian
+    /// - Value type: Unsigned
+    #[inline(always)]
+    pub fn multiplexor_raw(&self) -> u8 {
+        let signal = self.raw.view_bits::<Lsb0>()[0..4].load_le::<u8>();
+
+        signal
+    }
+
+    pub fn multiplexor(&mut self) -> MultiplexTestMultiplexor {
+        match self.multiplexor_raw() {
+            0 => MultiplexTestMultiplexor::M0(MultiplexTestMultiplexorM0 { raw: self.raw }),
+            1 => MultiplexTestMultiplexor::M1(MultiplexTestMultiplexorM1 { raw: self.raw }),
+            _ => unreachable!(),
```

Comment:
This can be reached by user-provided values -- I think we should return a `CanError::InvalidMultiplexor { message_id: 200 }` here.

Reply:
Good point, just fixed it.